R/build-home-index.R

```diff
@@ -64,13 +64,17 @@ data_home_sidebar <- function(pkg = ".") {
 
   if (length(html_path)) {
     if (!file.exists(html_path)) {
-      abort(
-        sprintf(
-          "Can't find file '%s' specified by %s.",
-          pkg$meta$home$sidebar$html,
-          pkgdown_field(pkg, c("home", "sidebar", "html"))
-        )
+      rel_html_path <- fs::path_rel(html_path, pkg$src_path)
+
+      msg_fld <- pkgdown_field(
+        pkg, c('home', 'sidebar', 'html'), fmt = TRUE, cfg = TRUE
       )
+
+      cli::cli_abort(c(
+        "Can't locate {.file {rel_html_path}}.",
+        x = paste0(msg_fld, " is misconfigured.")
```

Comment:
`pkgdown_field()` now returns glue-formatted strings (`pkgdown_field('a', fmt = TRUE)` returns `{.field a}`) that are filled in by `cli_abort()`, we have to use paste0 because of the nested interpolation.

Reply:
FWIW that seems a little confusing to me

Reply:
I can fix. What is confusing, the way I implemented `pkgdown_field()`, or the way we currently have to use it?

Here's the issue. We could take a `glue` dependency, which might be handy now that everything uses `cli`.

``` r
library(cli)
library(glue)

msg <- '{.var variable}'

# nested interp - doesn't work 
cli::cli_abort('assign {msg}')
#> Error:
#> ! assign {.var variable}
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort("assign {msg}")
#>  2.   └─rlang::abort(...)

# works - paste0
cli::cli_abort(paste0('assign ', msg))
#> Error:
#> ! assign `variable`
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(paste0("assign ", msg))
#>  2.   └─rlang::abort(...)
              
# works - glue
cli::cli_abort(glue('assign {msg}'))
#> Error:
#> ! assign `variable`
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(glue("assign {msg}"))
#>  2.   └─rlang::abort(...)
```

<sup>Created on 2023-12-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Hmmmm, yeah, maybe this is the least worst approach. Let's leave it for now and I'll learn to live with it :)